qgrad/qutip.py

```diff
@@ -0,0 +1,45 @@
+"""
+QuTiP functions that work with Jax
+"""
+
+from jax import numpy as jnp
+
+
+def fidelity(a, b):
+    """
+    Computes fidelity between two kets.
+    
+    Args:
+        a (`:obj:numpy.array`): State vector (ket)
+        b (`:obj:numpy.array`): State vector (ket)
+        
+    Returns:
+        float: fidelity between the two state vectors
+    """
+    return jnp.abs(jnp.dot(jnp.transpose(jnp.conjugate(a)), b))
+
+
+def rot(params):
+    """
+    Returns a unitary matrix describing rotation around Z-Y-Z axis
+
+    Args:
+        params (`:obj:numpy.array`[float]): an array of three parameters defining the
+                                     rotation
+
+    Returns:
+        `:obj:numpy.array`[complex]: a 2x2 matrix defining the unitary
+    """
+    phi, theta, omega = params
+    cos = jnp.cos(theta / 2)
+    sin = jnp.sin(theta / 2)
+
+    return jnp.array(
+        [
+            [
+                jnp.exp(-0.5j * (phi + omega)) * cos,
+                -(jnp.exp(0.5j * (phi - omega))) * sin,
+            ],
```

Comment:
AFAIK, PEP8 encourages spaces between operators of lowest precedence

Reply:
I suppose but I used Black to auto format which is also now in PSF so we can just let it deal with the linting in the most default format.